provider/azure/azure_private_dns_test.go

```diff
@@ -325,8 +407,6 @@ func TestAzurePrivateDNSApplyChanges(t *testing.T) {
 	testAzurePrivateDNSApplyChangesInternal(t, false, &recordsClient)
 
 	validateAzureEndpoints(t, recordsClient.deletedEndpoints, []*endpoint.Endpoint{
-		endpoint.NewEndpoint("old.example.com", endpoint.RecordTypeA, ""),
-		endpoint.NewEndpoint("oldcname.example.com", endpoint.RecordTypeCNAME, ""),
```

Comment:
The test case is a bit invalid, as there won't ever be `DNSName`s in `UpdateOld` that aren't also in `UpdateNew`. But that is a preexisting issue.

Reply:
The test cases could use some more work indeed, should I file a follow-up issue?